tests/composition/pec/test_parameterestimationcomposition.py

```diff
@@ -120,6 +158,11 @@
 @pytest.mark.composition
 @pytest.mark.parametrize("inputs_dict, error_msg", run_input_test_args)
 def test_pec_run_input_formats(inputs_dict, error_msg):
+
+    # Need to convert from mapping proxy
+    # to dict to make the test pass.
+    inputs_dict = {k:v for k,v in inputs_dict.items()}
```

Comment:
Why is this necessary? can't pec.run handle general mapping type?

Reply:
Ok, so PEC was modifying the inputs dict in place, I have made it so that it generates a copy now. I was also having issues with Composition._parse_input_dict not handling mappingproxytype well because it checks if type() == dict. Copying the dict gets around this as well.